src/v/kafka/server/snc_quota_manager.cc

```diff
@@ -310,8 +315,212 @@ ss::future<> snc_quota_manager::quota_balancer_step() {
 }
 
 ss::future<> snc_quota_manager::quota_balancer_update() {
```

Comment:
this needs a comment above that explain what exactly is happening here.

Reply:
a comment is added in the header - pls let me know if that is not what you've proposed